### PR TITLE
[FLINK-31494][table-planner] Introduce SqlNodeConverter for SqlToOperationConverter

### PR DESCRIPTION
**Pull request by wuchong**
## What is the purpose of the change

Similar to [FLINK-31368](https://issues.apache.org/jira/browse/FLINK-31368), the  `SqlToOperationConverter` is a bit bloated. This PR is aiming to introduce `SqlNodeConverter` and move conversion logic from `SqlToOperationConverter` in the following PRs. 

## Brief change log

- Rename `SqlToOperationConverter` into `SqlNodeToOperationConversion`, because the Converter name may confuse developers regarding the relationship with `SqlNodeConverter` 
- Introduce `SqlNodeConverter` interface and make `SqlToOperationConverter` supports it.
- Migrate the conversion logic of `SqlCreateCatalog` to verify the happy path. 

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
@luoyuxia @xuzhiwen1255 , would be appreciated if you could help to review.

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlNodeToOperationConversion.java [hunk 1]
```diff
@@ -285,11 +287,18 @@ public static Optional<Operation> convert(
...
-        } else if (validated instanceof SqlDropCatalog) {
+
+        // delegate conversion to the registered converters first
+        SqlNodeConvertContext context = new SqlNodeConvertContext(flinkPlanner, catalogManager);
+        Optional<Operation> operation = SqlNodeConverters.convertSqlNode(validated, context);
+        if (operation.isPresent()) {
+            return operation;
+        }
+
+        // TODO: all the below conversion logic should be migrated to SqlNodeConverters
```
Comment (xuzhiwen1255) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlNodeToOperationConversion.java:298]: Do all sqlnodes need to implement?

**Comment by xuzhiwen1255** [comment]
LGTM.  Thank you @wuchong for contribution.

**Comment by wuchong** [comment]
Thank you for the review @xuzhiwen1255 . Merging...

**Pull request merged**